### ISSUE TITLE
provide a way for addons to state that their section should apply to  only base config

### ISSUE THREAD

**Issue by derekriemer** [nvaccess/nvda#7467]
STR:
do this in the python console.
import config;config.conf.spec["nb1"]={"sharksBite":'string("default="painfully")'}
config.conf["nb1"]
That works.
Now try this.
config.conf.spec["nb2"]={"dogsBark":'string(default="loudly")'}
config.conf.BASE_ONLY_SECTIONS.add("nb2")
config.conf["nb2"]
You get a key error.

This is a problem for addons such as tip of the day (refactor coming soon), crash hero, and other addons which want to make sure their addon config is oblivious to profile switches. This might be necessary if I.E. an addon uses config to store the placemarkers config for where to look for user supplied placemarkers or such.

I propose either fixing this bug, or in the case this is a real optimization, provide an addon public API to let users add sections for exclusion.

**Comment by leonardder** [comment]
I think it should even be considered to have  add-ons configuration sections in BASE_ONLY_SECTIONS by default, unless specified otherwise.

**Comment by derekriemer** [comment]
I'd disagree. Things like unspoken shouldn't need to do extra work to get profiles.

**Comment by Adriani90** [comment]
cc: @feerrenrut

**Comment by nvdaes** [comment]
I agree with @derekriemer . I think that by default configuration for add-on should be profile independent.
I think that this needs to be triaged. Recently, this has been discussed on this [topic of the add-ons mailing list](https://nvda-addons.groups.io/g/nvda-addons/topic/119184212#msg27652).
Also, I have released placeMarkers 52.0.0 due to a bug reported with this add-on due to the addition of a section to BASE_ONLY_SECTIONS and then reset the configuration so that the configuration can be applied. The reported mentioned that, if he changes the state of screen curtain, then saves the configuration and ther resets NVDA, the the state of screen curtain is not keep. In addition, he reported that NVDA seems to be restarted perhaps slower.
cc: @seanbudd @SaschaCowley

**Comment by nvdaes** [comment]
I have thought about creating a method for config.conf named addSection, which may accept a dictionary and a boolean indicating if the section would be for the normal configuration or not. This should perform tasks like validation and adding to the spec.
If this is triaged, I may create a PR unless Derek or Leonard prefer to take this.

**Comment by nvdaes** [comment]
I'm making tests and exploring relevant code.
I think that an easy solution would be to try creating an [addon] section in the configspec file, including it in base only sections to validate it directly when the config manager is started. Then, when an add-on developer wants to add a section applied to the base configuration, they may use: config.conf.addSection(baseOnly=True, sectionName="myAddonId", spec=someDictionary)
By default, baseOnly would be False, and in this case the section would be added to config.conf directly, not to config.conf.getConfigValue("addons"...)

If add-on developers don't manifest concerns, I would try this approach. They may say that they don't want that a section created by them is inside an addon section, but I think that it may be more robust than trying to validate an undefined number of sections.

**Comment by CyrilleB79** [comment]
Thanks @nvdaes for trying to tackle this one.

If possible, I'd prefer a solution where add-ons "myAddon" keep its config in `config.conf["myAddon"]`, since it's from far the most used pattern.
There should then be a way to specify that section `[myAddon]` only applies to base profile.

If it is not suitable or doable from a technical point of view, feel free to ignore this request; but in this case, please explain the technical limitation in the PR, thanks.

**Comment by nvdaes** [comment]
Cyrille wrote:

> If possible, I'd prefer a solution where add-ons "myAddon" keep its config in config.conf


Thank you Cyrille, I'll try to do my best to achieve it. I'm seeing the following issues: since config is initialized before add-ons, and base only sections are validated then, for now I haven't been able to provide a suitable way to address this. But I'll be testing more.

**Comment by seanbudd** [comment]
I think you might have to register them via installTasks or a similar hook?

**Comment by nvdaes** [comment]
Sean wrote:

 I think you might have to register them via installTasks or a similar hook?

Great! I didn't think on this. I tested in the Python console with several approaches.
Then we can save added base only sections in a baseOnlySections.yaml file, so they can be recognized as base only when config is loaded, and config.conf.addSection may be use on installTask/onInstall function. I'll try this approach.

**Comment by nvdaes** [comment]
I'm working on this.
I had thought about raising errors if someone tries to add a section who is found in the spec or in base only sections, but now I think that it's better to remove this caution, since, if an add-on is updated and it wants to change something in a previously added section, this may cause undesirable issues.
We have add-ons making conflicts, for example, with gestures, and perhaps it's better to address conflicting sections separately, or leave it without been addressed for now.
Also, I think that a function to remove sections is also needed, perhaps in the same or in a followup PR.
I've used Copilot to create tests, refining them and trying to avoid new imports. And perhaps there are too many added tests, in particular, for a function that converts specs to be serialized in a yaml file, used to store the base only sections.
I'll perform more manual tests, and I plan to update the description of the PR and mark it for review, unless you want to provide comments before.
